javascript/ql/src/semmle/javascript/security/dataflow/ExceptionXss.qll

```diff
@@ -70,15 +67,18 @@ module ExceptionXss {
    */
   class Callback extends DataFlow::FunctionNode {
     DataFlow::ParameterNode errorParameter;
+    IfStmt ifStmt;
 
     Callback() {
       exists(DataFlow::CallNode call | call.getLastArgument().getAFunctionValue() = this) and
       this.getNumParameter() = 2 and
       errorParameter = this.getParameter(0) and
-      exists(Expr errorCheck |
-        isExplicitConditional(this.getFunction().getBodyStmt(0), errorCheck) and
-        errorParameter.flowsTo(DataFlow::valueNode(errorCheck))
-      )
+      ifStmt = this.getFunction().getBodyStmt(0) and
+      errorParameter.flowsToExpr(ifStmt.getCondition()) and
+      not ifStmt.getThen().getBasicBlock().getASuccessor*() = this
```

Comment:
@max-schaefer @asgerf it has apparently been too long since I looked at our basic block reasoning. I believe this implementation works, but I really would have liked to see a simpler way to express that a branch is terminal, but I must admit that I can not come up with a convincing alternative. Do we have another way of expressing terminal branches?

Reply:
I don't think we do, but I wonder whether this wouldn't be cleaner to express as a recursive predicate over the AST? (More verbose for sure, but the whole set-up seems fairly syntactic, so a syntactic criterion might be a better fit.)